src/uct/ib/rc/verbs/rc_verbs_iface.c

```diff
@@ -88,21 +88,30 @@ uct_rc_verbs_iface_poll_tx(uct_rc_verbs_iface_t *iface)
     ret = ibv_poll_cq(iface->super.super.send_cq, num_wcs, wc);
     if (ucs_unlikely(ret <= 0)) {
         if (ucs_unlikely(ret < 0)) {
+            /* Treat this as a local SW failure. Therefore do not
+             * move ep to the failed state. */
             ucs_fatal("Failed to poll send CQ");
         }
         return;
     }
 
     for (i = 0; i < ret; ++i) {
-        if (ucs_unlikely(wc[i].status != IBV_WC_SUCCESS)) {
-            ucs_fatal("Send completion with error: %s", ibv_wc_status_str(wc[i].status));
+        ep = ucs_derived_of(uct_rc_iface_lookup_ep(&iface->super, wc[i].qp_num),
+                            uct_rc_verbs_ep_t);
+
+        if ((ucs_unlikely(wc[i].status != IBV_WC_SUCCESS) || (ep == NULL))) {
```

Comment:
Once the fail occurs we remove the ep from iface array by means of uct_rc_iface_remove_ep. But the poll function may return several completions in a row, so we can get a bunch of send completions for the same ep. If the first completion fails we remove the ep from iface array and will not manage to find this ep anymore.